Release notes for 8.5.1 (#14745)
elastic/logstash: docs/static/releasenotes.asciidoc

*Json Codec - 3.1.1*

* Fix: when decoded JSON includes an `[event][original]` field, having `ecs_compatibility` enabled will no longer overwrite the decoded field https://github.com/logstash-plugins/logstash-codec-json/pull/43[#43]

*Grok Filter - 4.4.3*

* Minor typos in docs examples https://github.com/logstash-plugins/logstash-filter-grok/pull/176[#176]

*Tcp Input - 6.3.1*

* Fixes a regression in which the ssl_subject was missing for SSL-secured connections in server mode https://github.com/logstash-plugins/logstash-input-tcp/pull/199[#199]

*Unix Input - 3.1.2*

* Fix: eliminate high CPU usage when data timeout is disabled and no data is available on the socket https://github.com/logstash-plugins/logstash-input-unix/pull/30[#30]

*Rabbitmq Integration - 7.3.1*

* DOCS: clarify the availability and cost of using the `metadata_enabled` option https://github.com/logstash-plugins/logstash-integration-rabbitmq/pull/52[#52]

*Elasticsearch Output - 11.9.3*

> * DOC: clarify that `http_compression` option only affects _requests_; compressed _responses_ have always been read independent of this setting [#1030 ](https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/1030)

Reviewer: ```suggestion
* DOC: clarify that `http_compression` option only affects _requests_; compressed _responses_ have always been read independent of this setting https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/1030[#1030]
```

Reviewer: Looks like extra space after PR number confused the conversion script. I can't test this in comment form, but this should work.

* Fix broken link to Logstash Reference https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/1085[#1085]

* Fixes a possible infinite-retry-loop that could occur when this plugin is configured with an `action` whose value contains a [sprintf-style placeholder][] that fails to be resolved for an individual event. Events in this state will be routed to the pipeline's [dead letter queue][DLQ] if it is available, or will be logged-and-dropped so that the remaining events in the batch can be processed https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/1080[#1080]
 
[sprintf-style placeholder]: https://www.elastic.co/guide/en/logstash/current/event-dependent-configuration.html#sprintf

> [DLQ]: https://www.elastic.co/guide/en/logstash/current/dead-letter-queues.html

Reviewer: These links are acting up, and I'm not sure what the intention is. Check out the preview for easier viewing: 
https://logstash_14745.docs-preview.app.elstc.co/guide/en/logstash/8.5/logstash-8-5-1.html

If it's not obvious to you either, I can take a crack at it. 

If

> * Fixes a possible infinite-retry-loop that could occur when this plugin is configured with an `action` whose value contains a [sprintf-style placeholder][] that fails to be resolved for an individual event. Events in this state will be routed to the pipeline's [dead letter queue][DLQ] if it is available, or will be logged-and-dropped so that the remaining events in the batch can be processed https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/1080[#1080]
>  
> [sprintf-style placeholder]: https://www.elastic.co/guide/en/logstash/current/event-dependent-configuration.html#sprintf
> [DLQ]: https://www.elastic.co/guide/en/logstash/current/dead-letter-queues.html
> 

Reviewer: ```suggestion
* Fixes a possible infinite-retry-loop that could occur when this plugin is configured with an `action` whose value contains a <<sprintf,sprintf-style placeholder>> that fails to be resolved for an individual event. 
Events in this state are routed to the pipeline's <<dead-letter-queues,dead letter queue (DLQ)>> if the DLQ is enabled.
Otherwise, these events are logged-and-dropped so that the remaining events in the batch can be processed. https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/1080[#1080]
```
